src/CourseAuthoringPage.jsx

```diff
@@ -19,17 +23,28 @@ export default function CourseAuthoringPage({ courseId, children }) {
   const courseNumber = courseDetail ? courseDetail.number : null;
   const courseOrg = courseDetail ? courseDetail.org : null;
   const courseTitle = courseDetail ? courseDetail.name : courseId;
+  const apiCallStatus = useSelector(getCourseAppsApiStatus);
+  const inProgress = useSelector(getLoadingStatus) === RequestStatus.IN_PROGRESS;
+  if (apiCallStatus.status === 403) {
+    return (
+      <PermissionDeniedAlert />
+    );
+  }
+
+  const renderHeader = () => (
+    <Header
+      courseNumber={courseNumber}
+      courseOrg={courseOrg}
+      courseTitle={courseTitle}
+      courseId={courseId}
+    />
+  );
 
   return (
     <>
-      <Header
-        courseNumber={courseNumber}
-        courseOrg={courseOrg}
-        courseTitle={courseTitle}
-        courseId={courseId}
-      />
+      {inProgress ? <Loading /> : renderHeader()}
       {children}
```

Comment:
I am not sure, I have not dug deeper but why show children in In progress state

Reply:
Because CourseAppsAPI is being called inside the child component which changes `inprogress` and `CourseAppsApiStatus` states.